base/ltcmdhooks.dtx

```diff
@@ -76,8 +88,8 @@
 %
 %    The hooks are not physically present before
 %    \verb=\begin{document}= (i.e., using a command in the preamble
```

Comment:
I would suggest to add here that commands in the aux-file are not seeing the hooks either.

Reply:
Done, thanks.

This begs the question: is the timing right, or should the commands be patched earlier?  Or should we have a command `\hook_cmd_add_even_if_before_begindocument_because_I_need_it_in_the_aux:n`?

Reply:
Interesting question, but I think for now it might be better to leave it as that and just document the fact (which you did and I misread). If we really see a compelling need we might revisit but on the whole I think it make more confusion that it could gain. After all if we change it would either need to be a special "request" which is odd or we would need to move all code earlier and that doesn't quite work as package may add hooks in begin document

Reply:
@FrankMittelbach Yes, not for now, definitely.  The problem with `cmd` hooks and `\begin{document}` is not the hooks added `\AtBeginDocument` (they are patched in, as long as the command exists), but the commands defined there:  that's why we delayed back then, and it's likely hard to do it any other way.